File: Convert Sorted Array to Binary Search Tree/Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] num) {
        return sortedArrayToBST(num, 0, num.length - 1);
    }
    private TreeNode sortedArrayToBST(int[] num, int s, int e) {
        if (s > e) return null;
        int mid = (s + e) / 2;
        TreeNode node = new TreeNode(num[mid]);
        node.left = sortedArrayToBST(num, s, mid - 1);
        node.right = sortedArrayToBST(num, mid + 1, e);
        return node;
    }
}

File: Convert Sorted Array to Binary Search Tree/Convert Sorted Array to Binary Search Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode *sortedArrayToBST(vector<int> &num) {
        return sortedArrayToBST(num, 0, num.size() - 1);
    }
    TreeNode *sortedArrayToBST(vector<int> &num, int s, int e) {
        if (s > e) return NULL;
        int mid = (s + e) / 2;
        TreeNode *node = new TreeNode(num[mid]);
        node->left = sortedArrayToBST(num, s, mid - 1);
        node->right = sortedArrayToBST(num, mid + 1, e);
        return node;
    }
};